standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -5095,14 +5095,18 @@ public Partition alterPartition(String catName, String dbname, String name, List
       Iterator<List<String>> part_val_itr = part_vals.iterator();
       Set<MColumnDescriptor> oldCds = new HashSet<>();
       Ref<MColumnDescriptor> oldCdRef = new Ref<>();
+      MTable table = null;
       for (Partition tmpPart: newParts) {
         List<String> tmpPartVals = part_val_itr.next();
         if (writeId > 0) {
           tmpPart.setWriteId(writeId);
         }
         oldCdRef.t = null;
+        if (table == null) {
+          table = this.getMTable(tmpPart.getCatName(), tmpPart.getDbName(), tmpPart.getTableName());
+        }
         Partition result = alterPartitionNoTxn(
-            catName, dbname, name, tmpPartVals, tmpPart, queryWriteIdList, oldCdRef);
```

Comment:
Do we need the alterPartitionNoTxn with the following signature anymore

private Partition alterPartitionNoTxn(String catName, String dbname, String name,
      List<String> part_vals, Partition newPart, String validWriteIds, Ref<MColumnDescriptor> oldCd)

Even single alter partition can use the same call with the table already passed. Will not result to any improvement but just a clean up of the code.